### PR TITLE
arbiter: Support replica 2 arbiter 1 usage

### LINKED ISSUE #515: Arbiter volume creation syntax needs to change to 'replica 2 arbiter 1'
The  feedback from the community is that it would be better to use `replica 2 arbiter 1` instead of `replica 3 arbiter 1` in the volume create/ display syntax. See http://lists.gluster.org/pipermail/gluster-users/2017-December/033092.html

### PR DESCRIPTION
**Pull request by prashanthpai**
Change syntax for creation of arbiter volumes from:
`replica 3 arbiter `
to
`replica 2 arbiter 1`

Closes #515 
Signed-off-by: Prashanth Pai <ppai@redhat.com>

### CODE REVIEW

**Comment by amarts** [comment]
I guess it is OK. Lets not keep the backward compatibility, if that is not the correct behavior. I did it in GD1 as it would be required for some time. But GD2 being a new project replacing GD1 later (at least not in next immediate release), we should be good here.

**Comment by itisravi** [comment]
Okay, in that case it is fine.

File: glustercli/cmd/volume-create.go [hunk 1]
```diff
@@ -166,26 +166,44 @@ func volumeCreateCmdRun(cmd *cobra.Command, args []string) {
...
+		if numBricks%(flagCreateReplicaCount+flagCreateArbiterCount) != 0 {
+			failure("Invalid number of bricks specified. Number of bricks must be a multiple of replica (+arbiter) count.", nil, 1)
 		}
 
-		numSubvols := numBricks / flagCreateReplicaCount
+		// FIXME: Things would've been so much simpler if specifying the
+		// arbiter brick was simply a separate command flag such as
+		// --arbiter-brick or even a boolean as we always pick the last
+		// brick as arbiter. Backward compatibility is a pain and we're
+		// doing the same bad choices all over again.
```
Comment (aravindavk) [inline glustercli/cmd/volume-create.go]: I think it is not only for backward compatibility, but it is for future extension. In future, if we support `replica 3` and `arbiter 2`. 

@itisravi please add your thoughts.

**Comment by prashanthpai** [comment]
retest this please

**Comment by rishubhjain** [comment]
retest this please

**Comment by prashanthpai** [comment]
@aravindavk Can you review new changes in `glusterd2/bricksplanner/subvoltype_replicate.go` and `e2e/smartvol_ops_test.go` ?

File: glusterd2/bricksplanner/subvoltype_replicate.go [hunk 2]
```diff
@@ -22,7 +22,7 @@ func (s *replicaSubvolPlanner) Init(req *api.VolCreateReq, subvolSize uint64) {
 }
 
 func (s *replicaSubvolPlanner) BricksCount() int {
-	return s.replicaCount
+	return s.replicaCount + s.arbiterCount
 }
 
 func (s *replicaSubvolPlanner) BrickSize(idx int) uint64 {
```
Comment (aravindavk) [inline glusterd2/bricksplanner/subvoltype_replicate.go]: similar change required here as well

Comment (prashanthpai) [reply glusterd2/bricksplanner/subvoltype_replicate.go]: Thank you. Done :+1:

**Pull request merged**